# Send event about parsing XSL content
# Hacky way to get info about RSS content for such pages like
# http://newsrss.bbc.co.uk/rss/newsonline_world_edition/front_page/rss.xml
Index: mozilla/content/xml/document/src/nsXMLContentSink.cpp
===================================================================
--- mozilla.orig/content/xml/document/src/nsXMLContentSink.cpp
+++ mozilla/content/xml/document/src/nsXMLContentSink.cpp
@@ -772,16 +772,19 @@
                                    nsContentUtils::GetSecurityManager());
 
     NS_ENSURE_SUCCESS(rv, rv);
 
     if (NS_CP_REJECTED(decision)) {
       return NS_OK;
     }
 
+    nsContentUtils::DispatchTrustedEvent(mDocument,
+                                         static_cast<nsIContent*>(aElement),
+                                         NS_LITERAL_STRING("DOMLinkXSLParsed"), PR_TRUE, PR_TRUE);
     return LoadXSLStyleSheet(url);
   }
 
   // Let nsContentSink deal with css.
   rv = nsContentSink::ProcessStyleLink(aElement, aHref, aAlternate,
                                        aTitle, aType, aMedia);
 
   // nsContentSink::ProcessStyleLink handles the bookkeeping here wrt
